docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

```diff
@@ -4,17 +4,677 @@ title: Connector SDK
+validation from the connector function, use the `ConnectorContext` object's `validate` method
+as shown in the [runtime logic](#runtime-logic) section:
+
+```java
+...
+  @Override
+  public Object execute(ConnectorContext context) throws Exception {
+    ...
+    // (2)
+    context.validate(connectorRequest);
+    ...
+  }
+...
+```
+
+This will instruct the context to prepare a validator and pass it to the object that needs
+validation. To validate your input object `connectorRequest` using the API, it needs
+to implement the `ConnectorInput` interface. Using this interface's `validateWith` method,
+you can use the provided validator to ensure the data conforms to your requirements:
+
+```java
+package io.camunda.connector;
+
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.Validator;
+
+public class MyConnectorRequest implements ConnectorInput {
+
+  private String message;
+  private Authentication authentication;
+
+  @Override
+  public void validateWith(final Validator validator) {
+    validator.require(message, "message");
+    validateIfNotNull(authentication, validator);
+  }
+}
+```
+
+The validator contains convenience methods like `require` to allow for basic validations.
+Furthermore, the `ConnectorInput` interface comes with convenience methods to allow handling
+nested objects with `validateIfNotNull`. You can also integrate custom validations, adding
+custom errors to the validator's list of errors.
+
+```java
+package io.camunda.connector;
+
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.SecretStore;
+import io.camunda.connector.api.Validator;
+import java.util.Objects;
+
+public class Authentication implements ConnectorInput {
+
+  private String user;
+  private String token;
+
+  @Override
+  public void validateWith(final Validator validator) {
+    validator.require(user, "user");
+    validator.require(token, "token");
+    if (token != null && !token.startsWith("xobx")) {
+      validator.addErrorMessage("Token must start with \"xobx\"");
+    }
+  }
+}
+```
+
+Using this approach, you can validate your whole input data structure with one initial call from
+the central connector function.
+
+#### Secrets
+
+Connectors that require confidential information to connect to external systems need to be able
+to manage those securely. As described in the
+[guide for creating secrets](../../../console/manage-clusters/manage-secrets.md), secrets can be
+controlled in a secure location and referenced in a connector's properties using a placeholder
+pattern `secrets.*`. To make this mechanism as robust as possible, secret handling comes with
+the connector SDK out of the box. That way, all connectors can use the same standard way of
+handling secrets in input data.
+
+The SDK allows replacing secrets in input data as late as possible to avoid passing them around
+in the environments that handle connector invocation. Therefore, we do not pass secrets into the
```

Comment:
Minor tweak:
Update text from "Therefore, we do not pass..."
to "We do not pass..."

Reply:
For my understanding. This is only due to make it sound better? Because the first sentence gives the reason why we *do not pass*.

Alternateively we could state:
` This is the reason why we do not pass...`

Or do you think this is anyhow crystal clear?

Reply:
For me stating the obvious to the user is a little redundant, and we can avoid it where possible. 
I would pick one word "Therefore" over 5 words "This is the reason why".